src/System.Buffers/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace System.Buffers
+{
+    [EventSource(
+        Guid = "20b30044-b729-457e-8dda-8b41b5dd02e6", 
+        Name = "System.Buffers.BufferPoolEventSource",
+        LocalizationResources = "FxResources.System.Buffers.SR")]
+    internal sealed class ArrayPoolEventSource : EventSource
+    {
+        internal readonly static ArrayPoolEventSource Log = new ArrayPoolEventSource();
+
+        private const EventTask Buffer = (EventTask)1;
+
+        [Event(1, Level = EventLevel.Informational, Opcode = EventOpcode.Start, Task = Buffer)]
+        internal void BufferRented(int bufferId, int bufferSize, int poolId, int bucketId) { WriteEvent(1, bufferId, bufferSize, poolId, bucketId); }
+
+        [Event(2, Level = EventLevel.Informational, Opcode = EventOpcode.Stop, Task = Buffer)]
+        internal void BufferReturned(int bufferId, int poolId) { WriteEvent(2, bufferId, poolId); }
+
+        [Event(3, Level = EventLevel.Informational)]
+        internal void PooledBufferCreated(int bufferId, int bufferSize, int poolId, int bucketId) { WriteEvent(3, bufferId, bufferSize, poolId, bucketId); }
+
+        [Event(4, Level = EventLevel.Warning, Opcode = EventOpcode.Start, Task = Buffer)]
+        internal void UnPooledBufferAllocated(int bufferId, int bufferSize, int poolId) { WriteEvent(4, bufferId, bufferSize, poolId); }
+
+        [Event(5, Level = EventLevel.Warning, Opcode = EventOpcode.Start, Task = Buffer)]
```

Comment:
If Enums are passed to an EventSource Write event, then the meta data (enum values) for that enum is stored along with the other type information associated with the EventSource.   The public/private-ness in the program does not matter.   

I would NOT use 'Start' and 'Stop' Opcode designations for these events (just drop them).   When you use these opcode, EventSource will try to define an activity ID associated with it, and that semantics is extra overhead, and will probably cause grief in this case.    

As far as using an enum or not:   This is less important, but just ask yourself how you would like things presented if you were PROCESSING the event.   Odds ar you would want it present as a 'alloc' and 'free' where the 'alloc' might have special semantics (because you do the same thing for all three cases and then a small amount of per-variation logic).  This is especially nice if you have to 'remember' this bit while the allocation is outstanding, since you will need to 'seralize' the enum anyway.    THus I do recommend using an enum.